### PR TITLE
ML: Unable to select all media library assets, when page is greater than 1

### PR DESCRIPTION
**Pull request by gitstart**
### What does it do?

Fixes the issue with the inability to select all `Media Library `assets, when the page is greater than 1.

### Why is it needed?
When we attempt to select all in a media library with no folders, the `folder` variable being `undefined`, throws an error when we try to access `folders.map` 

### How to test it?

1. Go to Media Library
2. Upload a couple of assets such that you have multiple pages
3. Try to use the select all checkbox on the second or any page number which isn't 1
4. Verify that it selects all the items?

 
### Demos

https://user-images.githubusercontent.com/7220167/206154858-894ca9bb-525a-411e-87a3-d26acd41f14d.mov

### Related issue(s)/PR(s)

- Fixes https://github.com/strapi/strapi/issues/15062

### CODE REVIEW

**Comment by gitstart** [comment]
Hi @gu-stav, this PR is ready for review. kindly take a look.

File: packages/core/upload/admin/src/pages/App/MediaLibrary.js [hunk 1]
```diff
@@ -211,7 +211,7 @@ export const MediaLibrary = () => {
                       }
                       selectAll([
                         ...assets.map((asset) => ({ ...asset, type: 'asset' })),
-                        ...folders.map((folder) => ({ ...folder, type: 'folder' })),
+                        ...(folders ?? []).map((folder) => ({ ...folder, type: 'folder' })),
```
Comment (joshuaellis) [inline packages/core/upload/admin/src/pages/App/MediaLibrary.js:214]: Think this should be declared as early on as possible L85 for example. Would also reduce the optional chaining in the component too.

**Review by joshuaellis — CHANGES_REQUESTED** [review]

File: packages/core/upload/admin/src/pages/App/MediaLibrary.js (hunk 1 above)
Comment (gitstart) [reply packages/core/upload/admin/src/pages/App/MediaLibrary.js]: Thanks @joshuaellis, that works.

**Comment by joshuaellis** [comment]
So now we have a default, I would look at removing the optional chaining used in that file imo since we know `folders` is always defined as an array?

**Review by joshuaellis — APPROVED** [review]

**Comment by gitstart** [comment]
This PR was pushed through [Gitstart](https://www.gitstart.com/), with contributions from @oxenprogrammer, @raph941, @phunguyenmurcul.

**Pull request merged**